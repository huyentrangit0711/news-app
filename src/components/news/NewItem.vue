<template>
	<div class="new-item shadow-2xl rounded">
		<img class="object-cover h-48 w-full" :src="article.urlToImage" />
		<div class="content p-10">
			<h1 class="text-xl">{{ article.title }}</h1>
			<p class="description">
				{{ article.description }}
			</p>
		</div>
	</div>
</template>

<script>
export default {
	name: 'NewItem',
	props: ['article'],
	components: {},
};
</script>

<style></style>
