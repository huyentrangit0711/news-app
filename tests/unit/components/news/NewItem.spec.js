import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import NewItem from '@/components/news/NewItem';
const localVue = createLocalVue();
localVue.use(VueRouter);
const router = new VueRouter();
describe('News Item Component', () => {
	let wrapper;
	it('Component renders correctly', () => {
		wrapper = shallowMount(NewItem, {
			propsData: {
				article: {
					id: 12435,
					title: 'Mock title',
					description: 'Mock description',
				},
			},
			router,
			stubs: ['router-link'],
		});
		expect(wrapper).toMatchSnapshot();
	});
	it('Computed property newDetailLink should work correctly', () => {
		wrapper = shallowMount(NewItem, {
			propsData: {
				article: {
					id: 12435,
					title: 'Mock title',
					description: 'Mock description',
				},
			},
			router,
			stubs: ['router-link'],
		});
		expect(wrapper.vm.newDetailLink).toEqual({
			name: 'new-detail',
			params: { newId: 12435 },
		});
	});
});
