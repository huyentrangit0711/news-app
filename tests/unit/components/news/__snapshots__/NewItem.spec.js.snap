// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Item Component Component renders correctly 1`] = `
<div class="new-item shadow-2xl rounded">
  <router-link-stub to="[object Object]"><img class="object-cover h-48 w-full">
    <div class="content p-10">
      <h1 class="text-xl">Mock title</h1>
      <p class="description">
        Mock description
      </p>
    </div>
  </router-link-stub>
</div>
`;
